script/lint_darwin_plugins.sh

```diff
@@ -0,0 +1,88 @@
+#!/bin/bash
+
+# This script lints and tests iOS and macOS platform code.
+
+# So that users can run this script from anywhere and it will work as expected.
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
+REPO_DIR="$(dirname "$SCRIPT_DIR")"
+
+source "$SCRIPT_DIR/common.sh"
+
+function lint_package() {
+  local package_name=$1
+  local platform=$2
+  local podspec_dir="$REPO_DIR/packages/$package_name/$platform/$package_name.podspec"
+  local failure_count=0
+
+  if [[ -f "$podspec_dir" ]]; then
+    echo "Linting $platform $package_name.podspec"
+
+    # Build as frameworks.
+    # This will also run any tests set up as a test_spec. See https://blog.cocoapods.org/CocoaPods-1.3.0.
+    pod lib lint "$podspec_dir" --allow-warnings --fail-fast --silent
```

Comment:
I would like to get rid of `--allow-warnings` and add `--analyze` but there are a lot of issues that can be addressed in another PR.  Here are the camera warnings, including the error analyzer warning:

```
    - WARN  | license: Missing license type.
    - WARN  | description: The description is equal to the summary.
    - WARN  | [iOS] keys: Missing primary key for `source` attribute. The acceptable ones are: `git, hg, http, svn`.
    - ERROR | [iOS] build_pod: Static Analysis failed. You can use `--verbose` for more information. You can use `--no-clean` to save a reproducible buid environment.
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:66:24: warning: 'AVCapturePhotoOutput' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:69:43: warning: 'AVCaptureResolvedPhotoSettings' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:78:7: warning: 'JPEGPhotoDataRepresentationForJPEGSampleBuffer:previewPhotoSampleBuffer:' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:77:19: warning: 'AVCapturePhotoOutput' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:163:52: warning: this block declaration is not a prototype [-Wstrict-prototypes]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:170:32: warning: 'AVCapturePhotoOutput' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:257:26: warning: 'AVCapturePhotoOutput' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:276:3: warning: 'AVCapturePhotoSettings' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:276:39: warning: 'AVCapturePhotoSettings' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:299:48: warning: 'AVCaptureSessionPreset3840x2160' is only available on iOS 9.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:300:41: warning: 'AVCaptureSessionPreset3840x2160' is only available on iOS 9.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:850:10: warning: block implicitly retains 'self'; explicitly mention 'self' to indicate this is intended behavior [-Wimplicit-retain-self]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:804:5: warning: 'AVCaptureDeviceDiscoverySession' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:804:58: warning: 'AVCaptureDeviceDiscoverySession' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:805:44: warning: 'AVCaptureDeviceTypeBuiltInWideAngleCamera' is only available on iOS 10.0 or newer [-Wunguarded-availability]
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:494:5: warning: Incorrect decrement of the reference count of an object that is not owned at this point by the caller
    - WARN  | [iOS] xcodebuild:  /Users/m/Projects/plugins/packages/camera/ios/Classes/CameraPlugin.m:805:44: warning: 
```

Reply:
Presumably turning on modules for everything will happen before adding this, since many plugins have such problems, so this will be a long-term project. Given that, I think it would be good to capture this in a bug and add a comment about it here ("# Add --analyze and remove --allow-warnings; see https://github.com/flutter/flutter/issues/#####")

Reply:
Good idea.
https://github.com/flutter/flutter/issues/41443
https://github.com/flutter/flutter/issues/41444